### PR TITLE
Issue #6307, interactive examples broken, Failed to execute measure

### PR DESCRIPTION
**Pull request by schalkneethling**
@peterbe This is part 1 of the items mentioned here:
https://github.com/mdn/kuma/issues/6307#issuecomment-573764829

Related to this error:

```
perf.654b849a6fd9.js:1 Uncaught DOMException: Failed to execute 'measure' on 'Performance': The mark 'interactive-editor-loading' does not exist.
    at Object.setMeasure (https://developer.mozilla.org/static/build/js/perf.654b849a6fd9.js:1:703)
    at handlePerfMarks (https://developer.mozilla.org/static/build/js/perf.654b849a6fd9.js:1:948)
    at perfMsgHandler (https://developer.mozilla.org/static/build/js/perf.654b849a6fd9.js:1:1557)
setMeasure @ perf.654b849a6fd9.js:1
handlePerfMarks @ perf.654b849a6fd9.js:1
perfMsgHandler @ perf.654b849a6fd9.js:1
perf.654b849a6fd9.js:1 Uncaught DOMException: Failed to execute 'measure' on 'Performance': The mark 'interactive-editor-loading' does not exist.
    at Object.setMeasure (https://developer.mozilla.org/static/build/js/perf.654b849a6fd9.js:1:703)
    at handlePerfMarks (https://developer.mozilla.org/static/build/js/perf.654b849a6fd9.js:1:948)
    at perfMsgHandler (https://developer.mozilla.org/static/build/js/perf.654b849a6fd9.js:1:1557)
setMeasure @ perf.654b849a6fd9.js:1
handlePerfMarks @ perf.654b849a6fd9.js:1
perfMsgHandler @ perf.654b849a6fd9.js:1
```

### CODE REVIEW

File: kuma/static/js/utils/perf.js [hunk 1]
```diff
@@ -47,10 +47,15 @@ mdn.perf = {
...
-            measureData.endMark
-        );
+        try {
+            performance.measure(
+                measureData.measureName,
+                measureData.startMark,
+                measureData.endMark
+            );
+        } catch (error) {
+            console.error('Error while setting performance measure: ', error);
```
Comment (peterbe) [inline kuma/static/js/utils/perf.js]:
```suggestion
            console.error(`Error while setting performance measure: ${error.toString()}`);
```
Your code is fine but I wanna take this opportunity to better understand Interactive Examples and to carefully see if this change fixes things.

**Comment by peterbe** [comment]
I'm starting to have doubts about this. The only point of using `try/catch` is if you want to prevent the browser to give up on the rest of the file. If you have:
```html
<script>throw new Error('something')</script>
<script src="foo.js"></script>
```
the browser will happily download, parse, and execute `foo.js` even though an unexpected error was thrown/crashed in the inline scripts. If you have this:

```html
<script>throw new Error('something')</script>
<script>alert('This will still run')</script>
```
If you run that, you'll get an error in your Browser Console **and** you'll get an Window Alert box that appears. I.e. the second `<script>` block will go ahead. 
See: https://innovative-newsboy.glitch.me/

In other words, `try/catch` help us for anything outside the performance measurement stuff. 

The only time you should use try/catch is when there's code that you want should run AFTER something can go wrong. E.g. 
```html
<script>
try {   throw new Error('something')  }
catch (err) { console.log('That sucked') }
alert("This will run")
</script>
```
See: https://materialistic-passenger-77sp7p6bl.glitch.me/

So, the point is, I see no point trying to suppress any crashes within that performance stuff. It's fundamentally wrong to use `try/catch` to hope that it'll make "other things" work better.

File: kuma/static/js/utils/perf.js (hunk 1 above)
Comment (schalkneethling) [reply kuma/static/js/utils/perf.js]: Also, I would love to use JS template strings, but our pipeline does not allow us at the moment. If you run `build-static` you will see it throws an error:

```
pipeline.exceptions.CompressorError: b"Parse error at -:57,26\n            console.error(`Error while setting performance measure: ${error.toSt\n                          ^\nSyntaxError: Unexpected character '`'\n    at JS_Parse_Error.get (eval at <anonymous> (/tools/node_modules/uglify-js/tools/node.js:27:1), <anonymous>:86:23)\n    at /tools/node_modules/uglify-js/bin/uglifyjs:384:40\n    at time_it (/tools/node_modules/uglify-js/bin/uglifyjs:620:15)\n    at /tools/node_modules/uglify-js/bin/uglifyjs:345:9\n    at Socket.<anonymous> (/tools/node_modules/uglify-js/bin/uglifyjs:601:13)\n    at Socket.emit (events.js:215:7)\n    at endReadableNT (_stream_readable.js:1184:12)\n    at processTicksAndRejections (internal/process/task_queues.js:80:21)\n"
```

**Comment by schalkneethling** [comment]
> It's fundamentally wrong to use try/catch to hope that it'll make "other things" work better.

For sure but, I did not do this just because I was hoping that it will solve the problem with interactive examples. I also did it so we have a little more context around the error when it happens inside our minified/uglified code.

**Comment by peterbe** [comment]
> so we have a little more context around the error when it happens inside our minified/uglified code.

What more would we find out? I would argue that the "natural crash" is better than attempting to put it into `console.error`.

**Comment by schalkneethling** [comment]
> > so we have a little more context around the error when it happens inside our minified/uglified code.
> 
> What more would we find out? I would argue that the "natural crash" is better than attempting to put it into `console.error`.

Fair enough, so you reckon we should leave this as is?

**Comment by peterbe** [comment]
That is my review. That I think we should close this.
